From 97db9221ea304221a35337fc80bd3bfb760399de Mon Sep 17 00:00:00 2001
From: Oleksii Bidnichenko <oleksii.bidnichenko@toradex.com>
Date: Mon, 13 Nov 2023 10:41:32 -0300
Subject: [PATCH] drm: bridge: cadence: lower mhdp hdmi bridge ddc/edid read
 delay

The MHDP HDMI bridge driver is implemented in such a way that mostly it does
not check the status of PHY read operations and continuing normal execution even
if the status of the previous read operation is an error. Such an implementation
reduces the boot process speed in case if the hardware setup has a special
configuration e.g. I2C bus used for the display mode configuration is disabled.
Due to a testing need to have a stable boot time, a low-level read timeout
(MAILBOX_TIMEOUT_US) is lowered from 5 seconds to 50 milliseconds.

Related-to: ELB-4132
Related-to: TEI-998
Signed-off-by: Oleksii Bidnichenko <oleksii.bidnichenko@toradex.com>

- Updated patch since now the 'MAILBOX_TIMEOUT_US' stays inside the cdns-mhdp.h file.

Upstream-Status: Inappropriate [other]

- This patch is related to Tezi only, where the boot fails when it is
  running on the FCT Tester.

Related-to: ELB-1064
Signed-off-by: Hiago De Franco <hiago.franco@toradex.com>
---
 drivers/gpu/drm/bridge/cadence/cdns-mhdp.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/cadence/cdns-mhdp.h b/drivers/gpu/drm/bridge/cadence/cdns-mhdp.h
index 8ad99eb8f86e..1707204dc961 100644
--- a/drivers/gpu/drm/bridge/cadence/cdns-mhdp.h
+++ b/drivers/gpu/drm/bridge/cadence/cdns-mhdp.h
@@ -187,7 +187,7 @@
 #define CDNS_MHDP_MAX_STREAMS   4
 
 #define MAILBOX_RETRY_US		1000
-#define MAILBOX_TIMEOUT_US		5000000
+#define MAILBOX_TIMEOUT_US		50000
 
 #define mhdp_readx_poll_timeout(op, addr, offset, val, cond, sleep_us, timeout_us)	\
 ({ \
-- 
2.42.0

